benchmark/bench.py

```diff
@@ -0,0 +1,519 @@
+
+
+def to_bytes(size):
+    if size.isdigit():
+        return int(size)
+    unit = size[-1]
+    if unit == "K":
+        return int(size[:-1]) * 1024
+    elif unit == "M":
+        return int(size[:-1]) * (1024**2)
+    elif unit == "G":
+        return int(size[:-1]) * (1024**3)
+    elif unit == "T":
+        return int(size[:-1]) * (1024**4)
+    elif unit == "P":
+        return int(size[:-1]) * (1024**5)
+    else:
+        raise ValueError("Given size is not correct.")
+
+
+def build_s5cmd_exec(old, new, local_dir):
+    run_cmd(
+        [
+            "git",
+            "-C",
+            local_dir,
+            "clone",
+            "https://github.com/peak/s5cmd.git",
+            "-q",
+        ]
+    )
+
+    clone_dir = os.path.join(local_dir, "s5cmd")
+
+    old = S5cmd(local_dir, clone_dir, "old", old)
+    new = S5cmd(local_dir, clone_dir, "new", new)
+    return old, new
+
+
+def create_bench_dir(bucket, prefix, local_path):
+    """
+    Create a benchmark directory with a unique name to specified local_path.
+    If no path is specified, create temporary directory using mkdtemp.
+    In both of those cases, use the same unique name to create remote path name.
+
+    :param bucket: remote bucket to be used to create remote_path
+    :param prefix: prefix to be used after specified bucket for remote_path
+    :param local_path: specify a path for temporary files to be created in your local. If empty,
+    use default temporary folder path of your device.
+    :returns:
+        - local_dir - created local_dir path as local_path/bench-unique_suffix
+        - remote_path - created remote_path as s3://bucket/prefix/unique_suffix
+
+    """
+    if local_path:
+        basename = "bench"
+        suffix = datetime.datetime.now().strftime("%y%m%d_%H%M%S")
+        tmp_dir = "_".join([basename, suffix])
+        if os.path.isdir(local_path):
+            os.chdir(local_path)
+        else:
+            raise NotADirectoryError(local_path)
+        os.chdir(local_path)
+        os.mkdir(tmp_dir)
+        os.chdir(tmp_dir)
+
+        local_dir = os.getcwd()
+        remote_path = f"s3://{bucket}/{prefix}/{suffix}"
+    else:
+        local_dir = mkdtemp(prefix=prefix)
+        idx = local_dir.rfind(prefix[-1]) + 1
+        remote_path = f"s3://{bucket}/{prefix}/{local_dir[idx:]}"
+    print(f"All the local temporary files will be created at {local_dir}")
+    print(f"All the remote files will be uploaded to {remote_path}")
+
+    return local_dir, remote_path
+
+
+def run_cmd(cmd):
+    process = subprocess.run(cmd, capture_output=True, text=True)
+    print(process.stdout, end="")
+    print(process.stderr, end="")
+    return process.stdout
+
+
+def cleanup(tmp_dir, temp_result_file_dir):
+    temp_summary = os.path.join(temp_result_file_dir, "detailed_summary.md")
+    if os.path.isfile(temp_summary):
+        os.remove(temp_summary)
+
+    shutil.rmtree(tmp_dir)
+
+
+if __name__ == "__main__":
+    raise SystemExit(main())
```

Comment:
Is terminating a program with raising an Exception a python best practice?

What is the difference with the suggested version below?
```suggestion
    main()
```

Reply:
In this case, there is no difference in between them. I will commit your suggestion 👍